src/highdicom/legacy/sop.py

```diff
@@ -25,384 +31,506 @@
+    def __init__(self, number_of_frames: int) -> None:
+        """
+        Parameters
+        ----------
+        number_of_frames: int
+            The perframe functional group sequence will have items equal to
+            the whole number of frames
+        """
+        super().__init__()
+        for i in range(0, number_of_frames):
+            item = Dataset()
+            self.append(item)
 
-    # Create the Pixel Data element of the mulit-frame image instance using
-    # native encoding (simply concatenating pixels of individual frames)
-    # Sometimes there may be numpy types such as ">i2". The (* 1) hack
-    # ensures that pixel values have the correct integer type.
-    mf_dataset.PixelData = b''.join([
-        (ds.pixel_array * 1).data for ds in sf_datasets
-    ])
 
-    return mf_dataset
+class ـSharedFunctionalGroup(DataElementSequence):
 
+    """A sequence class for shared functional group"""
 
-class LegacyConvertedEnhancedMRImage(SOPClass):
+    def __init__(self) -> None:
+        """Construncts a shared frame functional group holding only one item
 
-    """SOP class for Legacy Converted Enhanced MR Image instances."""
+        """
+        super().__init__()
+        item = Dataset()
+        self.append(item)
+
+
+class FrameSet:
+
+    """
+        A class containing the dicom frames that hold equal distinguishing
+        attributes to detect all perframe and shared dicom attributes
+    """
+
+    def __init__(self, single_frame_list: list,
+                 distinguishing_tags: list) -> None:
+        """
+        Parameters
+        ----------
+        single_frame_list: list
+            list of single frames that have equal distinguising attributes
+        distinguishing_tags: list
+            list of distinguishing attributes tags
+        """
+        self._frames = single_frame_list
+        self._distinguishing_attributes_tags = distinguishing_tags
+        tmp = [
+            tag_for_keyword('AcquisitionDateTime'),
+            tag_for_keyword('AcquisitionDate'),
+            tag_for_keyword('AcquisitionTime'),
+            tag_for_keyword('SpecificCharacterSet')]
+        self._excluded_fromperframe_tags =\
+            self.distinguishing_attributes_tags + tmp
+        self._perframe_tags: list = []
+        self._shared_tags: list = []
+        self._find_per_frame_and_shared_tags()
+
+    @property
+    def frames(self) -> List[Dataset]:
+        return self._frames[:]
+
+    @property
+    def distinguishing_attributes_tags(self) -> List[Tag]:
+        return self._distinguishing_attributes_tags[:]
+
+    @property
+    def excluded_from_perframe_tags(self) -> List[Tag]:
+        return self._excluded_fromperframe_tags[:]
+
+    @property
+    def perframe_tags(self) -> List[Tag]:
+        return self._perframe_tags[:]
+
+    @property
+    def shared_tags(self) -> List[Tag]:
+        return self._shared_tags[:]
+
+    @property
+    def SeriesInstanceUID(self) -> UID:
```

Comment:
If the class is not derived from `pydicom.dataset.Dataset`, use the Python naming convention for attributes -> `series_instance_uid`